### ISSUE TITLE
Add new exercise: error-handling

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/javascript#910]
[problem specification ](https://github.com/exercism/problem-specifications/tree/master/exercises/error-handling)| [description](https://github.com/exercism/problem-specifications/blob/master/exercises/error-handling/description.md) | ~canonical data~

You may find it helpful to follow [this checklist](https://github.com/exercism/javascript/blob/83feda2af955df1ef9f3a8080338f506f0fdf696/CONTRIBUTING.md#new-exercise) from the [CONTRIBUTING.md](https://github.com/exercism/javascript/blob/83feda2af955df1ef9f3a8080338f506f0fdf696/CONTRIBUTING.md) resource. There is _not yet_ any canonical data at moment of creating this issue. You may come up with test cases yourself or (recommended) check tracks that have already implemented this exercise for inspiration.

**Comment by A-O-Emmanuel** [comment]
HI @SleeplessByte can I have a go at this, can I be assigned to it

**Comment by SleeplessByte** [comment]
@A-O-Emmanuel yes. [You can look at these 10 tracks](https://exercism.org/exercises/error-handling) for an example of how this exercise could work.

**Comment by A-O-Emmanuel** [comment]
@SleeplessByte Alright, thank you

**Comment by A-O-Emmanuel** [comment]
Hi @SleeplessByte 
I am encountering an issue with testing,

I consistently get: TypeError: (0 , _errorHandling.processString) is not a function

when I run:  npx babel-node scripts/test

I have tried:
-  ES Modules (export / import) and commonJS (modul.exports / require) syntax
- Cleared Jest cache
- Confirmed paths are correct

Yet the error persists. It appears the test may not be picking up the export correctly

Could you please advise if there is a known issue with this exercise's test setup or guide me on how to debug further.

Thank you.

**Comment by SleeplessByte** [comment]
Hi there!

Neither [the README.md](https://github.com/exercism/javascript?tab=readme-ov-file#running-the-test-suite) nor the [CONTRIBUTING.md guide](https://github.com/exercism/javascript/blob/main/CONTRIBUTING.md#test) mention `babel-node`. This was removed [at the beginning of this year (about 7 months ago)](https://github.com/exercism/javascript/pull/2548/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055). 

Is it possible you have not yet updated your fork to the latest files?

**Comment by A-O-Emmanuel** [comment]
@SleeplessByte thank you, I will update it.

**Comment by A-O-Emmanuel** [comment]
@SleeplessByte I have created the pull request for the error-handling exercise, it was automatically closed by the bot, with the intruction to make a post about the pull request  in the discussion forum, which i did.

**Comment by SleeplessByte** [comment]
@A-O-Emmanuel no worries, I opened it https://github.com/exercism/javascript/pull/2732.

I also formatted the code for you here: https://github.com/exercism/javascript/pull/2732/commits/69a0b956ed3e41db789320f1da936160b6ec058c so pull before you continue :)

**Comment by A-O-Emmanuel** [comment]
Hi @SleeplessByte I've made new commits on the error-handling exercise, i think this approach is more straight forward

**Comment by SleeplessByte** [comment]
@A-O-Emmanuel that is great! Sorry I am a bit occupied right now but will check as soon as I have time <3

**Comment by A-O-Emmanuel** [comment]
Alright

**Comment by A-O-Emmanuel** [comment]
Hi @SleeplessByte I made some changes to #2732 I basically reverted to the old way i did and also implemented your comments, please check if it's okay, i tried console logging the error message and also throwing it in the catch block so that it can be tested.

**Comment by A-O-Emmanuel** [comment]
HI @SleeplessByte the javascript exercise, have you been able to review it, I've made the recommended changes and also added new tests. #2732

**Comment by A-O-Emmanuel** [comment]
HI @SleeplessByte I have added the changes that @Cool-Katt wanted me to add, can it be reviewed if there's more i need to do?
. #2732

**Comment by A-O-Emmanuel** [comment]
Hi hope you're well, @SleeplessByte, @Cool-Katt has approved the pull request, now waiting for your approval as well, so that it can be merged
